File: case-study/case-study/src/components/customer/Customer.js
export class Customer {
    constructor(id, name, birthDay, gender,
                identity, phoneNumber, email,
                customerType, address) {
        this._id = id;
        this._name = name;
        this._birthDay = birthDay;
        this._gender = gender;
        this._identity = identity;
        this._phoneNumber = phoneNumber;
        this._email = email;
        this._customerType = customerType;
        this._address = address;
    }


    get id() {
        return this._id;
    }

    set id(value) {
        this._id = value;
    }

    get name() {
        return this._name;
    }

    set name(value) {
        this._name = value;
    }

    get birthDay() {
        return this._birthDay;
    }

    set birthDay(value) {
        this._birthDay = value;
    }

    get gender() {
        return this._gender;
    }

    set gender(value) {
        this._gender = value;
    }

    get identity() {
        return this._identity;
    }

    set identity(value) {
        this._identity = value;
    }

    get phoneNumber() {
        return this._phoneNumber;
    }

    set phoneNumber(value) {
        this._phoneNumber = value;
    }

    get email() {
        return this._email;
    }

    set email(value) {
        this._email = value;
    }

    get customerType() {
        return this._customerType;
    }

    set customerType(value) {
        this._customerType = value;
    }

    get address() {
        return this._address;
    }

    set address(value) {
        this._address = value;
    }
}

File: case-study-API/case_module5/src/main/java/com/case_module5/model/Customer.java
package com.case_module5.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "customers")
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    @Column(name = "name", columnDefinition = "VARCHAR(255)", nullable = false)
    private String name;
    @Column(name = "birth_day", columnDefinition = "DATE", nullable = false)
    private String birthDay;
    @Column(name = "identity", columnDefinition = "VARCHAR(50)", nullable = false)
    private String identity;
    @Column(name = "phone_number", columnDefinition = "VARCHAR(50)", nullable = false)
    private String phoneNumber;
    @Column(name = "email", columnDefinition = "VARCHAR(100)", nullable = false)
    private String email;
    @Column(name = "address", columnDefinition = "VARCHAR(100)", nullable = false)
    private String address;

    @ManyToOne
    @JoinColumn(name = "gender", referencedColumnName = "id")
    private Gender gender;

    @ManyToOne
    @JoinColumn(name = "customer_type", referencedColumnName = "id")
    private CustomerType customerType;

    public Customer(String name, String birthDay, String identity, String phoneNumber,
                    String email, String address, Gender gender, CustomerType customerType) {
        this.name = name;
        this.birthDay = birthDay;
        this.identity = identity;
        this.phoneNumber = phoneNumber;
        this.email = email;
        this.address = address;
        this.gender = gender;
        this.customerType = customerType;
    }
}
